### PR TITLE
Update awssecret UpstreamAuthority plugin to use AWS SDK v2

### PR DESCRIPTION
**Pull request by rturner3**
Consolidate usage of AWS SDK in SPIRE to only v2.

### CODE REVIEW

**Comment by rturner3** [comment]
Keeping in draft state until I get a chance to test in AWS.

**Review by azdagron — APPROVED** [review]

File: pkg/server/plugin/upstreamauthority/awssecret/awslib.go [hunk 1]
```diff
@@ -34,32 +32,31 @@ func readARN(ctx context.Context, sm secretsManagerClient, arn string) (string,
...
-	}
-
-	if config.SecretAccessKey != "" && config.AccessKeyID != "" {
-		awsConfig.Credentials = credentials.NewStaticCredentials(config.AccessKeyID, config.SecretAccessKey, config.SecurityToken)
-	}
+func newSecretsManagerClient(ctx context.Context, cfg *Configuration, region string) (secretsManagerClient, error) {
+	var credsProvider aws.CredentialsProvider
+	switch {
+	case cfg.AssumeRoleARN != "":
+		stsConf := aws.Config{
```
Comment (azdagron) [inline pkg/server/plugin/upstreamauthority/awssecret/awslib.go:39]: Looking at the example [here](https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/credentials/stscreds#hdr-Assume_Role), I think we should construct this config by loading the default config, similar to the default case.

**Comment by rturner3** [comment]
Thank you @MarcosDY for testing these changes on AWS and iterating on this PR. Marking as ready for review since they have been verified.

File: pkg/server/plugin/upstreamauthority/awssecret/awslib.go (hunk 1 above)
Comment (rturner3) [reply pkg/server/plugin/upstreamauthority/awssecret/awslib.go]: Fixed in latest version

**Pull request merged**